specification/security_model.adoc

```diff
@@ -1,9 +1,195 @@
 [[security_model]]
-== Security and Threat Model
+== Security Model
+
+The CoVE-IO securiy model is built on the following assumptions:
+
+- Only the TVM owner can assess of a TDI trustworthiness. Based on that
+  assessment, it explicitly accepts or rejects a TDI into its TCB.
+- A TDI may access a TVM confidential memory only when the following conditions
+  are met:
+  1. The TVM owner has explicitly allowed the TDI to access its memory by
+     accepting it.
+  2. The TDI is exclusively assigned to the TVM, i.e. it must not be shared
+     other TVMs or any host software component.
+- A TVM may trust and access a TDI MMIO space if and only if the TDI is
+  exclusively assigned to the TVM.
+
+By means of the CoVE-IO guest ABI, TVMs are required to explicitly accept
+TDIs into their TCBs. Accepting or rejecting a TDI is a TVM specific decision,
+based on TVM specific set of verification policies and criteria. A TVM accepting
+a TDI does not imply that other TVMs on the same host platform would accept it
+as well.
+
+Multi-function, CoVE-IO-compliant devices may expose multiple TDIs assigned to
+different TVMs. It is the DSM responsibility to guarantee isolation between all
+assigned TDIs, on a per-TVM basis.
+
+TDIs must not be shared between TVMs, they must be exclusively assign to zero or
+at most one TVM.
+
+As a TVM TCB does not include PCIe switches and bridges, a selective PCIe IDE
+stream must be setup to guarantee end-to-end confidentiality and integrity
+protection between a TVM and a TDI. The host VMM is responsible for setting up a
+single selective IDE stream between the host platform and any physical device
+from which one or more TDIs are assigned to TVMs. All TDIs withing that physical
+device share the single PCIe IDE stream.
+
+The main security objective of the CoVE-IO security model is to protect a TVM's
+confidential data integrity and confidentiality while TDIs are assigned to it.
+At the same time, availability of those assigned TDIs is out of this model scope
+as e.g. the host VMM could remove them from the TVM at any time it sees fit. All
+TVM confidential data must be cleared and wiped before the host software stack
+can fully reclaim an assigned TDI.
+
+== Threat Model
+
+=== Assets
+
+The CoVE-IO security model aims at protecting the following assets:
+
+1. The TVM confidential data, which includes its confidential main memory, code
+   and state.
+2. Any assigned device I/O space that is mapped to the TVM address space,
+   also known as device MMIO.
+3. System memory.
```

Comment:
why "3. system memory" ?

Reply:
I mean that `host memory that is not assigned to a TVM`, as a TVM with assigned TDIs could potentially use the TDI to tamper with that part of host memory (See T004). Any advice on how we could define that asset?

Reply:
Even if the host memory is tampered, it won't impact the confidentiality or integrity of TVM.

I recommend we **only** focus on the asset for the confidential computing. Otherwise, we have too many things need to   consider and we may lose focus.

Reply:
Yes, I agree now. I removed that asset, and I removed the related threat.